esp/services/ws_workunits/ws_workunitsHelpers.cpp

```diff
@@ -1846,29 +1846,26 @@ unsigned WsWuInfo::getResourceURLCount()
     return 0;
 }
 
-void appendIOStreamContent(MemoryBuffer &mb, IFileIOStream *ios, bool forDownload, const char *outFile = nullptr)
+void WsWuInfo::readFileContent(IFile *source, const char* sourceFileName, MemoryBuffer &mb,
```

Comment:
you can avoid passing in both a IFileIO and a filename for the same file, by using IFileIO::queryFilename() for filename when required.

Or, you can pass in only the sourceFileName and have this method call creatIFile, then it can do the error checking (throwing an exception if doesn't exist) - see other comment.

Reply:
The 'sourceFileName' is not always the file name. I guess I should rename it to 'fileAlias'. For example, when the readFileContent() is called for a WU xml file, the 'sourceFileName' is 'WU archive eclxml'.